3rdparty/mshadow/mshadow/base.h

```diff
@@ -336,6 +336,8 @@ const float kPi = 3.1415926f;
   typedef int32_t index_t;
 #endif
 
+const index_t kInt32Limit = (int64_t{1} << 31) - 1;
```

Comment:
Can we add a comment to make it clear where is this being used.

Reply:
what's the advantage of manually defining this over using INT_MAX in <climits>?

Reply:
@Zha0q1 good suggestion !